src/java_bytecode/java_class_loader.h

```diff
@@ -41,8 +37,30 @@ class java_class_loadert:public messaget
 
   void load_entire_jar(java_class_loader_limitt &, const std::string &f);
 
+  void read_jar_file(java_class_loader_limitt &, const irep_idt &);
+
+  /// Given a \p class_name (e.g. "java.lang.Thread") try to load the
+  /// corresponding .class file by first scanning all .jar files whose
+  /// pathname is stored in \ref jar_files, and if that doesn't work, then scan
+  /// the actual filesystem using `config.java.classpath` as class path (INSTEAD
+  /// OF \ref java_cp_include_files!!). Returns a default-constructed parse tree
+  /// when unable to find the .class file.
+  java_bytecode_parse_treet &get_parse_tree(
+    java_class_loader_limitt &limit, const irep_idt &class_name);
+
+  /// Maps class names to the bytecode parse trees.
+  typedef std::map<irep_idt, java_bytecode_parse_treet> class_mapt;
+  class_mapt class_map;
+
+  /// Maps .jar filesystem paths to jar_filet objects (stored inside).
+  /// Responsible for loading new jar files when they are first referenced.
   jar_poolt jar_pool;
 
+  /// An object of this class represents the information of _a single JAR file_
+  /// that is relevant for a class loader: a map associating logical class names
+  /// (e.g. "FIXME") with with handles (struct entryt) that describe one .class
```

Comment:
:) Thanks, I had fixed this 2 commits later. I edited the commits to fix it directly on the first commit.